pkg/util/storage/manager.go

```diff
@@ -80,5 +87,22 @@ func (m *manager) BlobService(ctx context.Context, resourceGroup, account string
 
 	blobcli := azstorage.NewAccountSASClient(account, v, (*m.env.Environment()).Environment).GetBlobService()
 
-	return &blobcli, nil
+	return &wrappedStorageClient{&blobcli}, nil
+}
+
+type wrappedStorageClient struct {
+	client *azstorage.BlobStorageClient
+}
+
+func (c *wrappedStorageClient) GetContainerReference(name string) *azstorage.Container {
+	return c.client.GetContainerReference(name)
+}
+
+func (c *wrappedStorageClient) Get(uri string) (io.ReadCloser, error) {
```

Comment:
It would be kind to have this document here https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob?tabs=microsoft-entra-id
Are we sure that there is no query parameter in the end of the URI?

Reply:
It would be safe to use URI Parser instead.

Reply:
I've revamped this func